apnsworker/src/main/scala/apnsworker/Apns.scala

```diff
@@ -0,0 +1,42 @@
+package apnsworker
+
+import cats.effect._
+import cats.syntax.functor._
+import com.turo.pushy.apns.{ApnsClient => PushyApnsClient}
+import db.{RegistrationService, ShardRange, Topic}
+import _root_.models.{Notification, iOS}
+import fs2.Stream
+import models.ApnsConfig
+
+import scala.concurrent.{ExecutionContext, Future, blocking}
+
+class Apns[F[_]](registrationService: RegistrationService[F, Stream], config: ApnsConfig)
+  (implicit executionContext: ExecutionContext, contextShift: Concurrent[F], F: Async[F]) {
+
+  private val apnsClient: F[PushyApnsClient] =
+    ApnsClient(config).fold(e => F.raiseError(e), c => F.delay(c))
+
+  def send(notification: Notification, shardRange: ShardRange): Stream[F, Either[Throwable, String]] = {
+
+    def sendAsync(token: String)(client: PushyApnsClient): F[String] =
+      Async[F].async[String] { (cb: Either[Throwable, String] => Unit) =>
+        ApnsClient.sendNotification(token, notification)(cb)(client, config)
+      }
+
+    def tokens: Stream[F, String] = notification
+        .topic
+        .map(topic => registrationService.find(Topic(topic.name), iOS, shardRange))
```

Comment:
You're probably aware of it as this is a WIP, but we'll need to support either a sql request that de-duplicates when querying multiple topics, or an in-memory de-duplication.

Content notification and football notification have generally more than one topic at a time and if I understand the code correctly a single token might come up multiple times.

Reply:
Yes I have a `TODO` to let the db handle that. Merging stream is not adequate here